net/art/table.go

```diff
@@ -18,17 +18,27 @@ import (
 	"io"
 	"net/netip"
 	"strings"
+	"sync"
 )
 
 // Table is an IPv4 and IPv6 routing table.
 type Table[T any] struct {
-	v4 strideTable[T]
-	v6 strideTable[T]
+	v4       strideTable[T]
+	v6       strideTable[T]
+	initOnce sync.Once
+}
+
+func (t *Table[T]) init() {
+	t.initOnce.Do(func() {
+		t.v4.prefix = netip.PrefixFrom(netip.IPv4Unspecified(), 0)
+		t.v6.prefix = netip.PrefixFrom(netip.IPv6Unspecified(), 0)
+	})
```

Comment:
I'm pretty sure this allocates on on 2nd+ calls since the func closes over the receiver t.

using a method value wouldn't:

```go
func (t *Table[T]) init() {  t.initOnce.Do(t.doInit) }
func (t *Table[T]) doInit() {
  t.v4.prefix = netip.PrefixFrom(netip.IPv4Unspecified(), 0)
  t.v6.prefix = netip.PrefixFrom(netip.IPv6Unspecified(), 0)
}
```

Reply:
It doesn't _appear_ to allocate, benchmarks say there's 1 alloc per Get, and I believe it's the `bs := addr.AsSlice()` a few lines below there. But I'll refactor and verify.

Reply:
Did this refactor. Benchmarks show no change in allocs/op for any of insert/delete/get (still ~2 allocs/insert, 1 alloc/delete, 1 alloc/get). I don't mind keeping or discarding it, it seems to have no significant impact.

Reply:
Huh, cool. I'm curious now when this improved. I'd stick with your prettier code if it doesn't allocate.

Reply:
Ack, keeping original implementation.